### PR TITLE
[libc][math][c23] add entrypoints and tests for getpayload{,f,f128}

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: libc/spec/stdc.td [hunk 1]
```diff
@@ -733,7 +733,10 @@ def StdC : StandardSpec<"stdc"> {
 	  GuardedFunctionSpec<"totalordermagf16", RetValSpec<IntType>, [ArgSpec<Float16Ptr>, ArgSpec<Float16Ptr>], "LIBC_TYPES_HAS_FLOAT16">,
 	  GuardedFunctionSpec<"totalordermagf128", RetValSpec<IntType>, [ArgSpec<Float128Ptr>, ArgSpec<Float128Ptr>], "LIBC_TYPES_HAS_FLOAT128">,
 
-          GuardedFunctionSpec<"getpayloadf16", RetValSpec<Float16Type>, [ArgSpec<Float16Ptr>], "LIBC_TYPES_HAS_FLOAT16">,
+          FunctionSpec<"getpayload", RetValSpec<DoubleType>, [ArgSpec<DoublePtr>]>,
+          FunctionSpec<"getpayloadf", RetValSpec<FloatType>, [ArgSpec<FloatPtr>]>,
+	  GuardedFunctionSpec<"getpayloadf16", RetValSpec<Float16Type>, [ArgSpec<Float16Ptr>], "LIBC_TYPES_HAS_FLOAT16">,
+	  GuardedFunctionSpec<"getpayloadf128", RetValSpec<Float128Type>, [ArgSpec<Float128Ptr>], "LIBC_TYPES_HAS_FLOAT128">,
```
Comment (overmighty) [inline libc/spec/stdc.td:740]: Nit: you're mixing tabs and spaces.

```suggestion
          FunctionSpec<"getpayload", RetValSpec<DoubleType>, [ArgSpec<DoublePtr>]>,
          FunctionSpec<"getpayloadf", RetValSpec<FloatType>, [ArgSpec<FloatPtr>]>,
          GuardedFunctionSpec<"getpayloadf16", RetValSpec<Float16Type>, [ArgSpec<Float16Ptr>], "LIBC_TYPES_HAS_FLOAT16">,
          GuardedFunctionSpec<"getpayloadf128", RetValSpec<Float128Type>, [ArgSpec<Float128Ptr>], "LIBC_TYPES_HAS_FLOAT128">,

```

**Review by overmighty — APPROVED** [review]
LGTM besides a whitespace nit.

**Review by lntue — APPROVED** [review]

**Comment by mikhailramalho** [comment]
@Jobhdez I think this patch is causing an issue in rv32:
```
/usr/local/bin/clang++ -DLIBC_NAMESPACE=__llvm_libc_20_0_0_git -D_DEBUG -I/home/mgadelha/llvm-project/build1/projects/libc/test/src/math/smoke -I/home/mgadelha/llvm-project/libc/test/src/math/smoke -I/home/mgadelha/llvm-project/libc -isystem /home/mgadelha/llvm-project/build1/projects/libc/include -mabi=ilp32d -march=rv32imafdc --target=riscv32-unknown-linux-gnu --sysroot=/opt/riscv/sysroot --gcc-toolchain=/opt/riscv -fPIC -fno-semantic-interposition -fvisibility-inlines-hidden -Werror=date-time -Werror=unguarded-availability-new -Wall -Wextra -Wno-unused-parameter -Wwrite-strings -Wcast-qual -Wmissing-field-initializers -pedantic -Wno-long-long -Wc++98-compat-extra-semi -Wimplicit-fallthrough -Wcovered-switch-default -Wno-noexcept-type -Wnon-virtual-dtor -Wdelete-non-virtual-dtor -Wsuggest-override -Wstring-conversion -Wmisleading-indentation -Wctad-maybe-unsupported -fdiagnostics-color -g -std=c++17 --target=riscv32-unknown-linux-gnu -fpie -DLIBC_FULL_BUILD -ffreestanding -fno-exceptions -fno-unwind-tables -fno-asynchronous-unwind-tables -fno-rtti -ffixed-point -MD -MT projects/libc/test/src/math/smoke/CMakeFiles/libc.test.src.math.smoke.getpayloadf128_test.__unit__.__build__.dir/getpayloadf128_test.cpp.o -MF projects/libc/test/src/math/smoke/CMakeFiles/libc.test.src.math.smoke.getpayloadf128_test.__unit__.__build__.dir/getpayloadf128_test.cpp.o.d -o projects/libc/test/src/math/smoke/CMakeFiles/libc.test.src.math.smoke.getpayloadf128_test.__unit__.__build__.dir/getpayloadf128_test.cpp.o -c /home/mgadelha/llvm-project/libc/test/src/math/smoke/getpayloadf128_test.cpp
In file included from /home/mgadelha/llvm-project/libc/test/src/math/smoke/getpayloadf128_test.cpp:9:
/home/mgadelha/llvm-project/libc/test/src/math/smoke/GetPayloadTest.h:41:7: error: no viable conversion from 'BigInt<128, false>' to 'long double'
   41 |     T default_snan_payload = StorageType(1) << (FPBits::SIG_LEN - 2);
      |       ^                      ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/mgadelha/llvm-project/libc/test/src/math/smoke/getpayloadf128_test.cpp:13:1: note: in instantiation of member function 'GetPayloadTestTemplate<long double>::testNaNs' requested here
   13 | LIST_GETPAYLOAD_TESTS(float128, LIBC_NAMESPACE::getpayloadf128)
      | ^
/home/mgadelha/llvm-project/libc/test/src/math/smoke/GetPayloadTest.h:68:42: note: expanded from macro 'LIST_GETPAYLOAD_TESTS'
   68 |   TEST_F(LlvmLibcGetPayloadTest, NaNs) { testNaNs(&func); }
      |                                          ^
/home/mgadelha/llvm-project/libc/src/__support/big_int.h:430:56: note: explicit conversion function is not a candidate
  430 |   template <typename T> LIBC_INLINE constexpr explicit operator T() const {
      |                                                        ^
1 error generated.

**Comment by overmighty** [comment]
I'll fix it.

**Pull request merged**